### PR TITLE
initial switch logic for new drand network

### PR DESCRIPTION
**Pull request by CluEleSsUK**
## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: build/drand.go [hunk 1]
```diff
@@ -27,15 +27,18 @@ const (
 	DrandDevnet
 	DrandLocalnet
 	DrandIncentinet
+	DrandQuicknet
 )
 
 var DrandConfigs = map[DrandEnum]dtypes.DrandConfig{
 	DrandMainnet: {
+		//FUCK: add storswift endpoint
```
Comment (jennijuju) [inline build/drand.go:35]: lololol

**Comment by Kubuxu** [comment]
One concern I have with the current impl is that the "which round out of the 10" (30s / 3s = 10 possible rounds) is a valid round is a recursive property based on the previous round. I would prefer if it was fully defined in a non-recursive manner.

File: build/drand.go [hunk 2]
```diff
@@ -44,6 +47,22 @@ var DrandConfigs = map[DrandEnum]dtypes.DrandConfig{
...
+			"https://api3.drand.sh",
+			"https://drand.cloudflare.com",
+			//"https://api.drand.secureweb3.com:6875", // Storswift
+		},
+		Relays: []string{
+			"/dnsaddr/api.drand.sh/",
+			"/dnsaddr/api2.drand.sh/",
+			"/dnsaddr/api3.drand.sh/",
+		},
+		ChainInfoJSON: `{"public_key":"a0b862a7527fee3a731bcb59280ab6abd62d5c0b6ea03dc4ddf6612fdfc9d01f01c31542541771903475eb1ec6615f8d0df0b8b6dce385811d6dcf8cbefb8759e5e616a3dfd054c928940766d9a5b9db91e3b697e5d70a975181e007f87fca5e","period":3,"genesis_time":1677685200,"hash":"dbd506d6ef76e5f386f41c651dcb808c5bcbd75471cc4eafa3f4df7ad4e4c493","groupHash":"a81e9d63f614ccdb144b8ff79fbd4d5a2d22055c0bfe4ee9a8092003dab1c6c0","schemeID":"bls-unchained-on-g1","metadata":{"beaconID":"fastnet"}}`,
```
Comment (AnomalRoil) [inline build/drand.go:64]: These are the `fastnet` chain info, not the `quicknet` ones.
```suggestion
		ChainInfoJSON: `{"public_key":"83cf0f2896adee7eb8b5f01fcad3912212c437e0073e911fb90022d3e760183c8c4b450b6a0a6c3ac6a5776a2d1064510d1fec758c921cc22b0e17e63aaf4bcb5ed66304de9cf809bd274ca73bab4af5a6e9c76a4bc09e76eae8991ef5ece45a","period":3,"genesis_time":1692803367,"hash":"52db9ba70e0cc0f6eaf7803dd07447a1f5477735fd3f661792ba94600c84e971","groupHash":"f477d5c89f21a17c863a7f937c6a6d15859414d2be09cd448d4279af331c5d3e","schemeID":"bls-unchained-g1-rfc9380","metadata":{"beaconID":"quicknet"}}`,
```

File: chain/beacon/drand/drand_test.go [hunk 3]
```diff
@@ -37,3 +40,93 @@ func TestMaxBeaconRoundForEpoch(t *testing.T) {
...
+			name:           "smaller drand block time adds divisor",
+			filRoundTime:   30,
+			drandRoundTime: 3,
+			round:          5,
+			expected:       15,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			drandInfo := fmt.Sprintf(`{"public_key":"a0b862a7527fee3a731bcb59280ab6abd62d5c0b6ea03dc4ddf6612fdfc9d01f01c31542541771903475eb1ec6615f8d0df0b8b6dce385811d6dcf8cbefb8759e5e616a3dfd054c928940766d9a5b9db91e3b697e5d70a975181e007f87fca5e","period":%d,"genesis_time":1677685200,"hash":"dbd506d6ef76e5f386f41c651dcb808c5bcbd75471cc4eafa3f4df7ad4e4c493","groupHash":"a81e9d63f614ccdb144b8ff79fbd4d5a2d22055c0bfe4ee9a8092003dab1c6c0","schemeID":"bls-unchained-on-g1","metadata":{"beaconID":"fastnet"}}`, test.drandRoundTime)
```
Comment (AnomalRoil) [inline chain/beacon/drand/drand_test.go:118]: This is also `fastnet` instead of `quicknet`.

**Comment by CluEleSsUK** [comment]
wait... this just closed but i didn't click close hmmmmm

**Comment by jennijuju** [comment]
what happened lolol

**Comment by arajasek** [comment]
It's cuz you force-pushed to be identical to master (commit `fb3970a9583411c8045de98972fb425893b1821f`), so GitHub auto-closed it as the PR is now a no-op.

I'm guessing you meant to force-push to...something else?

**Pull request closed without merging**